File: src/main/java/me/itzisonn_/meazy/lexer/UnknownTokenException.java
package me.itzisonn_.meazy.lexer;

import me.itzisonn_.meazy.Registries;
import me.itzisonn_.meazy.lang.TextException;
import me.itzisonn_.meazy.lang.text.Text;

/**
 * Is thrown when {@link Registries#TOKENIZATION_FUNCTION} can't recognize token
 */
public class UnknownTokenException extends TextException {
    /**
     * @param lineNumber Line number
     * @param errorString Error string
     */
    public UnknownTokenException(int lineNumber, String errorString) {
        super(Text.translatable("meazy:lexer.unknown_token", lineNumber, errorString));
    }
}

File: src/main/java/me/itzisonn_/meazy/lexer/UnknownTokenException.kt
package me.itzisonn_.meazy.lexer

import me.itzisonn_.meazy.registry.Registries
import me.itzisonn_.meazy.text.TextException
import me.itzisonn_.meazy.text.translatable

/**
 * Is thrown when [Registries.TOKENIZATION_FUNCTION] can't recognize token
 *
 * @param lineNumber Line number
 * @param errorString Error string
 */
class UnknownTokenException(lineNumber: Int, errorString: String) : TextException(
    translatable("meazy:lexer.unknown_token", lineNumber, errorString)
)